// Test definitions. Do NOT edit this file!
#define CATCH_CONFIG_MAIN // defines main() automatically
#include <iostream>
#include "catch.hpp"
#include "main.hpp"
// tests for exercise 1
TEST_CASE("Ex1 findMin() ", "[example]")
{
	vector<int> number = {2, 1, 4, 4, 0, 4, 3, 3, 4, 0};
	int size = 10;
	int result, usernum;
	// srand(time(0));
	// makevector(number, size);
	printvector(number);
	usernum = 4;
	result = deleteone(number, usernum);
	INFO(" 4 should be deleted 4 times. Your return value is " << result);
	REQUIRE(result == 4);
	INFO(" Vector size is 6. Your size is " << number.size());
	REQUIRE(number.size() == 6);
	cout << "--------------------------------------------------\n";
	usernum = 0;
	result = deleteone(number, usernum);
	INFO(" 0 should be deleted 2 times. Your return value is " << result);
	REQUIRE(result == 2);
	INFO(" Vector size is 4. Your size is " << number.size());
	REQUIRE(number.size() == 4);
	cout << "--------------------------------------------------\n";
	usernum = 20;
	result = deleteone(number, usernum);
	INFO("The result should be 0 when the value is not found\n");
	REQUIRE(result == 0);
	REQUIRE(number.size() == 4);
	cout << "--------------------------------------------------\n";

	// REQUIRE(number[idx] == usernum);
}